profiling/src/internal/profile.rs

```diff
@@ -846,36 +819,37 @@ mod api_test {
 
         // The string table should have at least the empty string.
         assert!(!profile.strings.is_empty());
-        assert_eq!("", profile.get_string(StringId::ZERO));
     }
 
     #[test]
     fn reset_period() {
         /* The previous test (reset) checked quite a few properties already, so
          * this one will focus only on the period.
          */
-        let mut profile = provide_distinct_locations();
-
-        let period = (
-            10_000_000,
-            ValueType {
-                r#type: profile.intern("wall-time"),
-                unit: profile.intern("nanoseconds"),
-            },
-        );
-        profile.period = Some(period);
+        let sample_types: &[api::ValueType] = &[api::ValueType {
+            r#type: "wall-time",
+            unit: "nanoseconds",
+        }];
+        let period = api::Period {
+            r#type: sample_types[0],
+            value: 10_000_000,
+        };
+        let mut profile = Profile::new(SystemTime::now(), sample_types, Some(period));
 
         let prev = profile
             .reset_and_return_previous(None)
             .expect("reset to succeed");
-        assert_eq!(Some(period), prev.period);
 
         // Resolve the string values to check that they match (their string
         // table offsets may not match).
-        let (value, period_type) = profile.period.expect("profile to have a period");
-        assert_eq!(value, period.0);
-        assert_eq!(profile.get_string(period_type.r#type), "wall-time");
-        assert_eq!(profile.get_string(period_type.unit), "nanoseconds");
+        for (value, period_type) in [profile.period.unwrap(), prev.period.unwrap()] {
+            assert_eq!(value, period.value);
+            let strings = profile.strings.iter().collect::<Vec<_>>();
+            let r#type: &str = strings[period_type.r#type.to_offset()];
+            let unit: &str = strings[period_type.unit.to_offset()];
```

Comment:
Minor: It may be useful to introduce a test helper for doing lookups in the strings, as it looks like something we may need to use for more tests.